### PR TITLE
feat(engine): Wilds Route, 1280x720 viewport, UI layout fixes

### PR DESCRIPTION
**Pull request by gcko**
### Wilds Route (Gap 4.4 Phase A)
- Add Roothollow village (30x25 tiles) with Vessa NPC, herbalist shop, save point, and Scene 5 dialogue trigger (Torren joins the party)
- Add Maren's Refuge interior (16x14 tiles) with Maren NPC and Scene 6 dialogue trigger (Maren joins the party)
- Expand overworld with bidirectional transitions to both new locations + location labels for navigation
- Add PartyState.add_member() and has_member() public API for party assembly during gameplay
- Add dialogue_scene_id metadata support (loads dialogue from DataManager instead of inline metadata)
- Add required_flag metadata for prerequisite flag checks on dialogue triggers
- Party join deferred until dialogue overlay closes (via overlay_state_changed one-shot)
- Add Thornmere Wilds encounter zone to overworld data
- Create Roothollow herbalist shop (9 consumable items per economy.md)
- Extend tileset to 10 tiles (forest floor, bioluminescent accent)
- Add NPC ambient dialogue files for Vessa, Maren, and Roothollow herbalist (canonical text from npcs.md)

### Viewport Change (1280x720)
- Change native viewport from 320x180 to 1280x720
- Add 4x camera zoom to exploration scene (game world renders at pixel art scale)
- Scale ALL UI overlay scenes to 1280x720: dialogue, menu, save/load, shop, battle, title
- Scale all hardcoded pixel offsets in UI scripts (dialogue_box, menu_overlay, battle_ui, save_load, title)
- Fix pre-existing gdlint ordering issues in title.gd and save_load.gd

### UI Fixes
- FF6-style inline speaker names in dialogue ("SPEAKER: text" on same line, no separate label)
- Fix speaker name duplication bug on dialogue re-show (duplicate() lines array before prepending)
- Save point menu converted from absolute positioning to VBoxContainer (fixes Rest/Rest&Save overlap)
- Menu party rows converted to VBox+HBox container layout (fixes member info overlap)
- Save slot display restructured with HeaderRow HBox + PartyLabel (fixes slot info overlap)
- Menu command panel converted to VBoxContainer layout

### Infrastructure
- Add Godot --import step to pre-push hook with proper error capture (set +e guard)
- Add 2 new checklist items to verification-checklists.md (item removal grep, GDScript float division)
- Update gap 3.4 with Phase 3: FF6-style bordered panels for all 9 sub-screens (tracked as beads issue)
- Update gap tracker viewport references from 320x180 to 1280x720

### Gap 4.4 Remaining Phases
- **Phase A2:** Fenmother's Hollow dungeon (3 floors, 2 bosses, forest biome)
- **Phase B:** Opening Sequence (Ironmouth, Ember Vein F3, Scenes 1-4, Lira+Sable join)
- **Phase C:** Capital Completion (remaining Valdris districts, Scene 7, Thornwatch, Act I finale)

### Gap 3.4 Phase 3 (NEW — P1)
- FF6-style bordered panels for all 9 menu sub-screens (Item, Magic, Abilities, Equipment, Status, Formation, Config, Save, Ley Crystal)
- Tracked as beads issue pendulum-of-despair-afs

## Test plan
- [x] Pre-commit hooks pass (gdlint, gdformat, JSON validation)
- [x] Pre-push quality gates pass (ID uniqueness, stale counts, scene refs, GUT tests)
- [x] 396 GUT tests passing across 23 test files
- [x] Cross-reference tests validate NPC dialogue, shop items, transitions, spawn markers
- [x] Round-trip transition tests for overworld <-> roothollow and overworld <-> marens_refuge
- [x] Speaker name duplication regression test (test_speaker_name_no_duplication_on_reshow)
- [x] Shop item count validation (9 items, no equipment in herbalist)
- [x] add_member edge cases (empty id, negative level, duplicates, overflow to reserve)
- [x] Manual: Walked to Roothollow, verified Scene 5 dialogue fires
- [x] Manual: Verified speaker name format (FF6 inline style)
- [x] Manual: Verified menu party panel layout at 1280x720
- [x] Manual: Verified save screen slot display layout

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: game/scripts/core/exploration.gd [hunk 1]
```diff
@@ -351,13 +354,39 @@ func _on_dialogue_trigger_entered(body: Node2D, area: Area2D) -> void:
...
+
+
+func _check_party_joining_flags() -> void:
+	if EventFlags.get_flag("torren_joined") and not PartyState.has_member("torren"):
+		PartyState.add_member("torren", _get_party_avg_level())
+		flash_location_name("Torren joined the party!")
+	if EventFlags.get_flag("maren_warning") and not PartyState.has_member("maren"):
+		PartyState.add_member("maren", _get_party_avg_level())
+		flash_location_name("Maren joined the party!")
+
```
[the remark this answers is not in the document]
Comment (gcko) [reply game/scripts/core/exploration.gd]: Fixed in ad841fc. Party join check now deferred via one-shot overlay_state_changed connection. Flag is set immediately (to block re-entry), but add_member and flash fire only after dialogue overlay closes.

File: game/scripts/autoload/party_state.gd [hunk 2]
```diff
@@ -65,6 +65,30 @@ func initialize_new_game() -> void:
...
+	if formation.get("active", []).size() < 4:
+		formation["active"].append(idx)
+	else:
+		if not formation.has("reserve"):
+			formation["reserve"] = [] as Array[int]
+		formation["reserve"].append(idx)
+	var char_data: Dictionary = DataManager.load_character(character_id)
+	var default_row: String = char_data.get("default_row", "back")
+	formation["rows"][character_id] = default_row
+
```
[the remark this answers is not in the document]
Comment (gcko) [reply game/scripts/autoload/party_state.gd]: Fixed in 1e467be. add_member now checks prev_size vs members.size() after _add_character call and returns early if character was not added. Also validates empty character_id and clamps level >= 1.

File: docs/superpowers/specs/2026-04-08-wilds-route-design.md [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+(tile 1) borders. Bioluminescent accent tiles (tile 9) along paths.
+Town floor (tile 5) for walkable interior areas.
+
+NW area: Entrance from overworld
+  - Dialogue trigger zone for Scene 5 (first visit only)
+  - Flag-gated: fires only when !torren_joined
+
+Central chamber: Vessa NPC + Torren NPC
+  - Vessa: flag-gated dialogue (npc_vessa in dialogue data)
+  - Torren: present until torren_joined flag set
```
[the remark this answers is not in the document]
Comment (gcko) [reply docs/superpowers/specs/2026-04-08-wilds-route-design.md]: Fixed in 106b005. Clarified that Torren joins via Scene5Trigger dialogue, not as a placed NPC entity.

File: docs/superpowers/specs/2026-04-08-wilds-route-design.md [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+
+---
+
+## 1. Scope
+
+### In Scope
+
+- 2 new map scenes: Roothollow village, Maren's Refuge cottage
+- Overworld expansion: 2 new transition triggers + spawn markers
+- 1 new shop: Roothollow Herbalist (Act I consumables + unique gear)
```
[the remark this answers is not in the document]
Comment (gcko) [reply docs/superpowers/specs/2026-04-08-wilds-route-design.md]: Fixed in 5c97056. Changed 'unique gear' to 'consumables' in spec.

File: game/assets/tilesets/placeholder_dungeon.tres [hunk 5]
```diff
@@ -13,6 +13,8 @@ texture_region_size = Vector2i(16, 16)
 5:0/0 = 0
 6:0/0 = 0
 7:0/0 = 0
+8:0/0 = 0
+9:0/0 = 0
```
[the remark this answers is not in the document]
Comment (gcko) [reply game/assets/tilesets/placeholder_dungeon.tres]: The 8:0/0 = 0 and 9:0/0 = 0 entries in the .tres are Godot's atlas tile registration format — they declare tiles at atlas x-positions 8 and 9 (different columns in the 160x16 strip). The /0 = 0 means 'alternative 0, terrain set 0'. This is the same format used for tiles 0-7 and is correct.

File: game/tests/test_wilds_route.gd [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+	var shop: Dictionary = data.get("shop", data)
+	var inv: Array = shop.get("inventory", [])
+	for item: Variant in inv:
+		if item is Dictionary:
+			var item_id: String = (item as Dictionary).get("item_id", "")
+			assert_false(
+				item_id == "despair_ward",
+				"herbalist should not stock equipment (despair_ward is in craftsman shop)",
+			)
+
```
[the remark this answers is not in the document]
Comment (gcko) [reply game/tests/test_wilds_route.gd]: Fixed in a07e96d. Renamed to test_roothollow_herbalist_excludes_despair_ward.

File: game/scripts/ui/save_load.gd [hunk 7]
```diff
@@ -129,28 +129,42 @@ func _refresh_slot_display() -> void:
...
 		var loc: String = raw_loc if raw_loc != "" else "Unknown"
 		var pt: int = data.get("meta", {}).get("playtime", 0)
-		header.text = (
-			"%s%s  %d:%02d  %dg" % [prefix, loc, pt / 3600, (pt % 3600) / 60, world.get("gold", 0)]
-		)
+		header.text = "%s%s" % [prefix, loc]
 		header.modulate = COLOR_NORMAL
+		if playtime_lbl:
+			playtime_lbl.text = "%d:%02d" % [pt / 3600, (pt % 3600) / 60]
+			playtime_lbl.modulate = COLOR_NORMAL
```
[the remark this answers is not in the document]
Comment (gcko) [reply game/scripts/ui/save_load.gd]: Fixed in eb0b528. Wrapped with int() for explicit float-to-int conversion.
